articles/connections/social/oauth2.md

```diff
@@ -16,121 +16,106 @@ useCase:
+- **Scope** - The `scope` parameters to send with the authroization request. Separate multiple scopes with spaces.
+- **Client ID** - The Client ID for Auth0 as an application used to request authorization and exchange the authorization code. You will need to register with the identity provider in order to get a Client ID.
+- **Client Secret** - The Client Secret for Auth0 as an application used to exchange the authorization code. You will need to register with the identity provider in order to get a Client Secret.
+- **Fetch User Profile Script** - A Node.js script to call a userinfo URL with the provided access token. Please see the section below for details on this script.
 
-The `access_token` parameter is used for authenticating requests to the provider's API.
+Once the custom connection is created, the **Applications** tab will be active where you can select when the Connection will appear.
 
-::: note
-We recommend using the field names from the [normalized profile](/users/normalized).
-:::
+## Fetch User Profile Script
 
-For example, the following code will retrieve the user profile from the **GitHub** API:
+The fetch user profile script is called after the user has logged in with the OAuth2 provider. Auth0 executes this script to call the OAuth2 provider API and get the user profile:
 
 ```js
-function(access_token, ctx, callback) {
-    var request = require('request');
-    request.get('https://api.github.com/user', {
-        'headers': {
-            'Authorization': 'Bearer ' + access_token,
-            'User-Agent': 'Auth0'
-        }
-    }, function (e, r, b) {
-        if (e) {
-            return callback(e);
-        }
-        if (r.statusCode !== 200) {
-            return callback(new Error('StatusCode:' + r.statusCode));
-        }
-        callback(null, JSON.parse(b));
-    });
+function fetchUserProfile(accessToken, context, callback) {
+  request.get(
+    {
+      url: 'https://auth.example.com/userinfo',
+      headers: {
+        'Authorization': 'Bearer ' + accessToken,
+      }
+    },
+    (err, resp, body) => {
+      if (err) {
+        return callback(err);
+      }
+
+      if (resp.statusCode !== 200) {
+        return callback(new Error(body));
+      }
+
+      let bodyParsed;
+      try {
+        bodyParsed = JSON.parse(body);
+      } catch (jsonError) {
+        return callback(new Error(body));
+      }
+
+      const profile = {
+        user_id: bodyParsed.account.uuid,
+        email: bodyParsed.account.email
+      };
+
+      callback(null, profile);
+    }
+  );
 }
 ```
 
-You can filter, add or remove anything from the profile returned from the provider. However, it is recommended that you keep this script as simple as possible. More sophisticated manipulation of user information can be achieved through the use of [Rules](/rules). One advantage of using Rules is that they apply to *any* connection.
-
-### Fetch user profile for OIDC-conformant OAuth2 providers
```

Comment:
Removed this both to dissuade using this for OIDC connections and because the ID token was being used without validation.